packages/prompts/src/autocomplete.ts

```diff
@@ -0,0 +1,287 @@
+								)
+							: '';
+
+					// Render options with selection
+					const displayOptions =
+						this.filteredOptions.length === 0
+							? []
+							: limitOptions({
+									cursor: this.cursor,
+									options: this.filteredOptions,
+									style: (option, active) => {
+										const label = getLabel(option);
+										const hint = option.hint ? color.dim(` (${option.hint})`) : '';
+
+										return active
+											? `${color.green(S_RADIO_ACTIVE)} ${label}${hint}`
+											: `${color.dim(S_RADIO_INACTIVE)} ${color.dim(label)}${hint}`;
+									},
+									maxItems: opts.maxItems,
+									output: opts.output,
+								});
+
+					// Show instructions
+					const instructions = [
+						`${color.dim('↑/↓')} to select`,
+						`${color.dim('Enter:')} confirm`,
+						`${color.dim('Type:')} to search`,
+					];
+
+					// No matches message
+					const noResults =
+						this.filteredOptions.length === 0 && valueAsString
+							? [`${color.cyan(S_BAR)}  ${color.yellow('No matches found')}`]
+							: [];
+
+					// Return the formatted prompt
+					return [
+						title,
+						`${color.cyan(S_BAR)}  ${color.dim('Search:')} ${searchText}${matches}`,
+						...noResults,
+						...displayOptions.map((option) => `${color.cyan(S_BAR)}  ${option}`),
+						`${color.cyan(S_BAR)}  ${color.dim(instructions.join(' • '))}`,
+						`${color.cyan(S_BAR_END)}`,
+					].join('\n');
+				}
+			}
+		},
+	});
+
+	// Return the result or cancel symbol
+	return prompt.prompt() as Promise<Value | symbol>;
+};
+
+// Type definition for the autocompleteMultiselect component
+export interface AutocompleteMultiSelectOptions<Value> {
+	/**
+	 * The message to display to the user
+	 */
+	message: string;
+	/**
+	 * The options for the user to choose from
+	 */
+	options: Option<Value>[];
+	/**
+	 * The initial selected values
+	 */
+	initialValues?: Value[];
+	/**
+	 * The maximum number of items that can be selected
+	 */
+	maxItems?: number;
+	/**
+	 * The placeholder to display in the input
+	 */
+	placeholder?: string;
+	/**
+	 * The stream to read from
+	 */
+	input?: NodeJS.ReadStream;
+	/**
+	 * The stream to write to
+	 */
+	output?: NodeJS.WriteStream;
+}
+
+/**
+ * Integrated autocomplete multiselect - combines type-ahead filtering with multiselect in one UI
+ */
+export const autocompleteMultiselect = <Value>(opts: AutocompleteMultiSelectOptions<Value>) => {
+	const formatOption = (option: Option<Value>, active: boolean, selectedValues: Value[]) => {
+		const isSelected = selectedValues.includes(option.value);
+		const label = option.label ?? String(option.value ?? '');
+		const hint = option.hint ? color.dim(` (${option.hint})`) : '';
+		const checkbox = isSelected ? color.green(S_CHECKBOX_SELECTED) : color.dim(S_CHECKBOX_INACTIVE);
+
+		if (active) {
+			return `${color.green('›')} ${checkbox} ${label}${hint}`;
+		}
+		return `${color.dim(' ')} ${checkbox} ${color.dim(label)}${hint}`;
```

Comment:
The `multiselect` prompt don't rely on `›` (which is a Unicode (1.1) char, so it should be handled by the `unicodeOr` function) to indicate which line is focused.

The `multiselect` use the fact that only the focused line is not `dim`